### ISSUE TITLE
Enhance unit deduction functionality by adding 'unit="1"' to some blocks

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#227]
**Reported by dietmarw on 26 Oct 2009 12:15 UTC**
Some time ago I was presented with a simple example model where apparently the unit deduction didn't work as expected in Dymola. The Dynasim people quickly presented me with a way how unit deduction can also be made to work in this case (see attached [attachment:UnitDeductTest.mo]). So all what was missing was to tell the tool that the unit of the (in this case) gain was "1". 

In the attached model, for example the integral of the current is calculated and the gain of the integrator is defined as:
```text/x-modelica
 parameter Real k=1 "Integrator gain";
```
the resultant output signal will be unitless (see `AsUnitless.y`). **But** if one changes the gain to the following:
```text/x-modelica
 parameter Real(unit="1") k=1 "Integrator gain";
```
(as done in `Asec`) the resulting signal will have the correct unit 'C'(='A.s').

Looking at the MSL I found several places (especially Multibody) where already `unit="1"` was added as parameter modifier.

So wouldn'd it be a good idea to add this change to blocks like:
```
Modelica.Blocks.Continous.{Integrator|LimIntegrator|Deriative|FirstOrder|Secondorder|PI|PID|LimPID} 
```
and maybe others. Or are there any drawbacks to such a change?

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/227

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 Jul 2010 07:19 UTC**
Fixed in af01213debfd82ab3e16d76346673466b4c6925b (all scalar gains in blocks of the Modelica.Blocks library have now a unit="1" definition. Also a test model for this is provided in ModelicaTest)

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 16 Jul 2010 07:54 UTC**
